python_modules/dagster/dagster/_core/definitions/graph_definition.py

```diff
@@ -521,7 +522,9 @@ def to_job(
         name: Optional[str] = None,
         description: Optional[str] = None,
         resource_defs: Optional[Mapping[str, ResourceDefinition]] = None,
-        config: Optional[Union[ConfigMapping, Mapping[str, object], "PartitionedConfig[T]"]] = None,
+        config: Optional[
```

Comment:
I'm slightly leery of changing our public type signatures. I think the only alternative is to have the type signatures "lie" and just use type: ignores at test callsites for now. Not sure which approach is better.

Reply:
This seems reasonable for now. Will cut that into a stacked PR we can merge later.